compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -177,6 +188,18 @@ macro_rules! ungated {
             name: sym::$attr,
             encode_cross_crate: $encode_cross_crate,
             type_: $typ,
+            safety: AttributeSafety::Normal,
+            template: $tpl,
+            gate: Ungated,
+            duplicates: $duplicates,
+        }
+    };
+    (unsafe $attr:ident, $typ:expr, $tpl:expr, $duplicates:expr, $encode_cross_crate:expr $(,)?) => {
```

Comment:
Maybe move the `unsafe` variant before the regular ones so that they are matched first. That should make it less likely that there is an accidental match where `unsafe` is matched as `$attr`.